common/network-common/src/main/java/org/apache/spark/network/util/ByteUnit.java

```diff
@@ -54,7 +54,7 @@ public double toBytes(long d) {
     if (d < 0) {
       throw new IllegalArgumentException("Negative size value. Size must be positive: " + d);
     }
-    return d * multiplier;
+    return d * (double)multiplier;
```

Comment:
We need a space after casts. I suppose the point is to avoid overflow when the result is going to be a double anyway, at the cost of precision maybe. I tested, and the precision issue won't matter, but then again, there's no reason this method should return `double`. There is no such thing as 0.5 bytes.

I think we can just fix that for Spark 3; its value is used as a `long` or `int` everywhere anyway.

While here, I think I'd also fix the expressions like `(long) Math.pow(1024L, 2L)` above. The multipliers should be `1L << 10`, `1L << 20`, ... `1L << 50`. There's again no point there in using floating point math.

Reply:
I realized that we need `return double` since `return long` causes a failure at [here](https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala#L229) regarding terabyte.

Reply:
Hm, really? what does it return? type-wise the numbers should easily fit into a long. 2 TiB = 2199023255552 bytes. I'm missing where the failure is here.